### DISCUSSION TITLE
PowerBi = Access to the resource is forbidden | Ingestion folder empty | Storage account URL "InvalidUri"

### DISCUSSION THREAD

**Discussion by psilantropy** [microsoft/finops-toolkit#558] [Q&A] [answered]
Hi all,

First time hub deployment and I am running into an issue, which I suspect might be network related, but this is my first time with a service like this, so not certain. No issues with the CM connector, just the FinOps hub.

Note my account is owner across the tenant, so I'm pretty sure this is not a permissions issue, but that's how it presents with some errors.

**PowerBI**
After downloading the CostSummary sample report and setting up, I am met with this error when Applying Changes

> Load
> CostDetails
> Access to the resource is forbidden.
> Query parameters

![Error Summary](https://github.com/microsoft/finops-toolkit/assets/79955883/0554a59c-fcba-4202-a03e-827f175dfb30)

**Storage Account**

ingestion folder is **empty**.

msexports has 3 days of successful imports in this format /msexports/<long id string>/FinOps_Export_SubName/20240101-20240131/
Which contains (file names amended for post)
```
FinOps_Export_Subname_1.csv
FinOps_Export_Subname_2.csv
FinOps_Export_Subname_3.csv
```
$logs is empty.

If I open the storage account ingestion URL in a browser https://finopshub01xxxxxxxxxxxxx.dfs.core.windows.net/ingestion I receive this error
`{"error":{"code":"InvalidQueryParameterValue","message":"Value for one of the query parameters specified in the request URI is invalid.\nRequestId:xxxxxxxxxxxxxxxxxxxx\nTime:2024-01-23T22:00:15.5234437Z"}}`

**Data Factory**

msexports trigger is started and looks ok.
Linked Services > Storage has an error in the account selection method window;

> Error details
> Error code
> [DEDICATED_DPERRORCODE0](https://go.microsoft.com/fwlink/?linkid=2156271#error-code-dedicated_dperrorcode0)
> Details
> Operation returned an invalid status code 'Forbidden'

However if I click _Test connection_ it results in _Connection successful_.

Thanks all :)

**Comment by psilantropy** [comment]
My scope folder might be wrong for my msexports path, checking and fixing now.

Edit: This fixed my msexports > ingestion issue, but PowerBI permissions issue remains. Linked services > storage error still present.

**Comment by psilantropy** [comment]
Tested something simpler;

New/downloaded CostSummary powerbi report, get data, ADLSg2, URL > OK > **Details: "Access to the resource is forbidden."**

Account directly assigned Storage Blob Data Read (also has CM contrib + Billing Reader)

**Comment by psilantropy** [comment]
Managed to get rid of the permissions error, but now just have various issues on the PowerBI reports. I'll close and comment on existing issues.

**Reply by lundejd** [reply]
Care to share how you "got rid" of the permissions issues? I'm having same, would like to know what steps you took to remediate. Thanks!

**Reply by psilantropy** [reply, marked as answer]
Account directly assigned Storage Blob Data Reader.
